src/cpl.rs

```diff
@@ -324,30 +358,46 @@ impl FromStr for CslStringList {
     }
 }
 
-/// Convenience for creating a [`CslStringList`] from a slice of _key_/_value_ tuples.
-///
-/// # Example
-///
-/// ```rust, no_run
-/// use gdal::cpl::CslStringList;
-///
-/// let opts = CslStringList::try_from(&[("One", "1"), ("Two", "2"), ("Three", "3")]).expect("known valid");
-/// assert_eq!(opts.len(), 3);
-/// ```
-impl<const N: usize> TryFrom<&[(&str, &str); N]> for CslStringList {
-    type Error = GdalError;
-
-    fn try_from(pairs: &[(&str, &str); N]) -> Result<Self> {
+impl FromIterator<CslStringListEntry> for CslStringList {
+    fn from_iter<T: IntoIterator<Item = CslStringListEntry>>(iter: T) -> Self {
         let mut result = Self::default();
-        for (k, v) in pairs {
-            result.set_name_value(k, v)?;
+        for e in iter {
+            result.add_entry(&e).unwrap_or_default()
+        }
+        result
+    }
+}
+
+impl<'a> FromIterator<&'a str> for CslStringList {
+    fn from_iter<T: IntoIterator<Item = &'a str>>(iter: T) -> Self {
+        iter.into_iter()
+            .map(Into::<CslStringListEntry>::into)
+            .collect()
+    }
+}
+
+impl FromIterator<String> for CslStringList {
+    fn from_iter<T: IntoIterator<Item = String>>(iter: T) -> Self {
+        iter.into_iter()
+            .map(Into::<CslStringListEntry>::into)
+            .collect()
+    }
+}
+
+impl Extend<CslStringListEntry> for CslStringList {
+    fn extend<T: IntoIterator<Item = CslStringListEntry>>(&mut self, iter: T) {
+        for e in iter {
+            self.add_entry(&e).unwrap_or_default();
```

Comment:
Oof :pensive:.

Reply:
@lnicola Here's my rational as to why this might be OK.
Most of the time you're going to be extending one `CslStringList` with another. The only way that call can fail is if your insertion `String` has `\r`, `\n`, or most importantly `\0`. If `iter` comes from a `CslStringList` then that validation has already happened. 

However, if `iter` comes from `[CslStringListEntry]`, then you could presumably circumvent that check. I added [this comment](https://github.com/georust/gdal/pull/457/files#diff-735b4e8f56e7d6ad6abe40c99808df64a2b2e05b2b13f976303cefe2dca42affR399-R400) for that reason.

The obvious alternative means no `Extend`, and `TryFrom`s to cover what is probably a rare edge case. On balance, I'd rather have `Extend`.

Reply:
Exactly. OOM might be another case where it fails, but it's pretty unlikely.

Reply:
FWIW, Added additional documentation to `CslStringListEntry::new_{pair|flag}`.